public/scripts/extensions/tts/index.js

```diff
@@ -638,6 +638,22 @@ async function processTtsQueue() {
             : text.replaceAll('*', '').trim(); // remove just the asterisks
     }
 
+    if (extension_settings.tts.apply_regex) {
```

Comment:
If you remove the fallback below, combine the condition with `extension_settings.tts.regex_pattern` to not try to parse empty regex.

```suggestion
    if (extension_settings.tts.apply_regex && extension_settings.tts.regex_pattern) {
```